plugins/status.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const async = require('async');
+
+exports.register = function () {
+    this.outbound = require('./../outbound');
+    this.queue_dir = require('./../outbound/queue').queue_dir;
+}
+
+exports.hook_capabilities = function (next, connection) {
+
+    if (connection.remote.ip === '127.0.0.1' || connection.remote.ip === '::1') {
+        connection.capabilities.push('STATUS');
+    }
+
+    next();
+}
+
+exports.hook_unrecognized_command = function (next, connection, params) {
+    const self = this;
+
+    const result_process = function (err, result) {
+        if (err) return next(DENY, err.message);
+        return connection.respond(211, result ? JSON.stringify(result) : "null", () => next(OK));
+    };
+
+    if (params[0] !== 'STATUS') {
+        next();
+    }
+    else if (connection.remote.ip !== '127.0.0.1' && connection.remote.ip !== '::1') {
+        return next(DENY, "STATUS not allowed remotely");
+    }
+    else {
+        if (server.cluster && !params[1].match(/^QUEUE LIST/)) {
+            self.call_master(params[1], result_process);
+        }
+        else {
+            self.command_action(params[1], result_process);
+        }
+    }
+}
+
```

Comment:
Just an observation (no need to change): in the next 3 functions, you've done `const self = this`, but in none of the cases is it necessary. You can just use `this` within the switch blocks and the correct `this` will be resolved. Generally we only create references to `this` when a) an enclosing callback alters `this` or b) when providing a more obvious name to `this` makes it easier to humans to parse/grok what `this` is. Ex: within plugins we often do: `const plugin = this`.

Reply:
Yep, comming from "different" languages I am trying to use this right. Well, these are just leftover from debuging, updated...